### PR TITLE
isort all (imports)

### PR DESCRIPTION
**Pull request by haipinglu**
### Description
- Run `isort` on all files to sort the imports (skipped 4 files, changed 39 files, Flake8 checked).
- Switch on `isort` pre-commit and hope it can work well. Even if not, I should be able to fix it.
- I will try to fix some other pre-commits if feasible. 

These changes are light, affecting only the format of the imports so a quick check by you is much appreciate before merging. Thanks.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

File: examples/digits_dann_lightn/main.py [hunk 1]
```diff
@@ -3,24 +3,27 @@
...
 import argparse
+
 # import warnings
 # import sys
 import logging
+import os
 
-# sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../..')))
-
-from kale.utils.csv_logger import setup_logger  # np error if move this to later, not sure why
```
Comment (XianyuanLiu) [inline examples/digits_dann_lightn/main.py:14]: So this error doesn't exist after moving, does it?

File: examples/action_dann_lightn/main.py [hunk 2]
```diff
@@ -7,20 +7,20 @@
 import logging
 import os
 
-# import sys
-
-# sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../..")))
-
-from kale.utils.csv_logger import setup_logger  # np error if move this to later, not sure why
```
Comment (XianyuanLiu) [inline examples/action_dann_lightn/main.py:14]: So this error doesn't exist after moving, does it?

Comment (XianyuanLiu) [reply examples/action_dann_lightn/main.py:14]: Checked it. No error on Win10.

File: examples/digits_dann_lightn/main.py (hunk 1 above)
Comment (XianyuanLiu) [reply examples/digits_dann_lightn/main.py:14]: Checked. No error but a Numpy warning which seems not related to this change. That's fine.
`UserWarning: The given NumPy array is not writeable, and PyTorch does not support non-writeable tensors.`

**Review by XianyuanLiu — APPROVED** [review]
The sort is nice! No problem with computer vision codes. Many thank!

File: examples/digits_dann_lightn/main.py (hunk 1 above)
Comment (haipinglu) [reply examples/digits_dann_lightn/main.py:14]: > So this error doesn't exist after moving, does it?

The previous wrapping was due to length 88. This line is <120 so no problem after setting black to length 120. That being said, you raised a very good point. Now the imports are resorted. That kind of fix will not be allowed anymore. I will remove the comment. Glad to see you have tested that there is no error but just a numpy warning. Many thanks! Peer review really improves quality!

**Review by RaivoKoot — APPROVED** [review]

**Review by pz-white — APPROVED** [review]
LGTM for GraphML part, thanks for the import sort adjustment!

**Review by sz144 — APPROVED** [review]
The changes in mpca.py are okay.

File: examples/digits_dann_lightn/main.py (hunk 1 above)
Comment (XianyuanLiu) [reply examples/digits_dann_lightn/main.py:14]: Sorry to unresolve this, I find it is okay on my pc with Win10, but it raises an error on HPC. After reverting it, the error is solved.

```
Traceback (most recent call last):
  File "main.py", line 13, in <module>
    from examples.action_dann_lightn.config import get_cfg_defaults
ModuleNotFoundError: No module named 'examples'
All is done!
```

Comment (haipinglu) [reply examples/digits_dann_lightn/main.py:14]: @XianyuanLiu in your examples, you could import those local modules as `from config import get_cfg_defaults`, as in digit_dann_lightn We can talk tomorrow if this is till an issue. Thanks.

Comment (XianyuanLiu) [reply examples/digits_dann_lightn/main.py:14]: Solved by `pip install .`. Many thanks 💯 
FTR, [set & save environment variable in conda envs](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#saving-environment-variables).

Comment (haipinglu) [reply examples/digits_dann_lightn/main.py:14]: What is your second line of reply on "set & save environment variable in conda envs." for? I can't get it without context?

Comment (XianyuanLiu) [reply examples/digits_dann_lightn/main.py:14]: Sorry. It is a record. Before you told me to use `pip install .`, I have followed the link to add `pykale` root path to the environment variable in my conda env and solved the problem. It is not the best option but may be useful in some other situations.

**Pull request merged**